pkg/bucketeer/sdk.go

```diff
@@ -221,6 +253,49 @@ func (s *sdk) BoolVariation(ctx context.Context, user *user.User, featureID stri
 	return v
 }
 
+func (s *sdk) BoolVariationDetail(
+	ctx context.Context,
+	user *user.User,
+	featureID string,
+	defaultValue bool) model.EvaluationDetail[bool] {
+	evaluation, err := s.getEvaluation(ctx, user, featureID)
+	if err != nil {
+		s.logVariationError(err, "BoolVariationDetail", user.ID, featureID)
+		s.eventProcessor.PushDefaultEvaluationEvent(user, featureID)
+		return model.NewEvaluationDetail[bool](
+			featureID,
+			user.ID,
+			"",
+			0,
+			model.EvaluationReasonClient,
+			defaultValue,
+		)
+	}
+	variation := evaluation.VariationValue
+	v, err := strconv.ParseBool(variation)
+	if err != nil {
+		s.logVariationError(err, "BoolVariationDetail", user.ID, featureID)
+		s.eventProcessor.PushDefaultEvaluationEvent(user, featureID)
+		return model.NewEvaluationDetail[bool](
+			featureID,
+			user.ID,
+			evaluation.VariationID,
+			evaluation.FeatureVersion,
+			model.ConvertEvaluationReason(evaluation.Reason.Type),
```

Comment:
If there is no specific reason, the conversion should be done in the `NewEvaluationDetail` to keep this function private.